src/wallet/coincontrol.h

```diff
@@ -63,73 +63,56 @@ class CCoinControl
 
     CCoinControl();
 
-    bool HasSelected() const
-    {
-        return !m_preset_inputs.empty();
-    }
-
-    bool IsSelected(const COutPoint& output) const
-    {
-        return m_preset_inputs.count(output) > 0;
-    }
-
-    bool IsExternalSelected(const COutPoint& output) const
-    {
-        return m_external_txouts.count(output) > 0;
-    }
-
-    std::optional<CTxOut> GetExternalOutput(const COutPoint& outpoint) const
-    {
-        const auto ext_it = m_external_txouts.find(outpoint);
-        if (ext_it == m_external_txouts.end()) {
-            return std::nullopt;
-        }
-
-        return std::make_optional(ext_it->second);
-    }
-
-    void Select(const COutPoint& output)
-    {
-        m_preset_inputs.insert(output);
-    }
-
-    void SelectExternal(const COutPoint& outpoint, const CTxOut& txout)
-    {
-        m_preset_inputs.insert(outpoint);
-        m_external_txouts.emplace(outpoint, txout);
-    }
-
-    void UnSelect(const COutPoint& output)
-    {
-        m_preset_inputs.erase(output);
-    }
-
-    void UnSelectAll()
-    {
-        m_preset_inputs.clear();
-    }
-
-    void ListSelected(std::vector<COutPoint>& vOutpoints) const
-    {
-        vOutpoints.assign(m_preset_inputs.begin(), m_preset_inputs.end());
-    }
-
-    void SetInputWeight(const COutPoint& outpoint, int64_t weight)
-    {
-        m_input_weights[outpoint] = weight;
-    }
-
-    bool HasInputWeight(const COutPoint& outpoint) const
-    {
-        return m_input_weights.count(outpoint) > 0;
-    }
-
-    int64_t GetInputWeight(const COutPoint& outpoint) const
-    {
-        auto it = m_input_weights.find(outpoint);
-        assert(it != m_input_weights.end());
-        return it->second;
-    }
+    /**
+     * Returns true if there are pre-selected inputs.
+     */
+    bool HasSelected() const;
+    /**
+     * Returns true if the given output is pre-selected.
+     */
+    bool IsSelected(const COutPoint& output) const;
+    /**
+     * Returns true if the given output is selected as an external input.
+     */
+    bool IsExternalSelected(const COutPoint& output) const;
+    /**
+     * Returns the external output for the given outpoint if it exists.
+     */
+    std::optional<CTxOut> GetExternalOutput(const COutPoint& outpoint) const;
+    /**
+     * Pre-selects the given output.
+     * The output will be included in the transaction even if it's not the most optimal choice.
```

Comment:
nit: would remove the "pre-selects" term and go directly with something like "store the given outpoint."

I don't think that the "pre" term here provides information to readers. (the second paragraph is great to express what the process will do with the selected outpoint)

(same for the others methods as well).

Reply:
How about this?
```suggestion
     * Lock-in the selected output for spending.
     * The output will be included in the transaction even if it's not the most optimal choice.
```

Reply:
yeah, sounds good 👌🏼 .